Prompts Picker

thenamankumar opened the pull request:
closes: https://linear.app/sourcegraph/issue/SRCH-1264/surface-prompts-in-cmd-k-in-vscode-sorted-by-last-used

Adds quick pick for prompts with `alt+p` shortcut.

https://www.loom.com/share/18c0c54259e34704834d47544e02b2a3?sid=b112f153-48f2-4d4c-9a1c-0e248252ccf0

## Test plan

- alt + p
- search and select any prompt
- it should be executed the same way it does from the prompt library. 

## Changelog

Adds quick pick for prompts with `alt+p` shortcut.

taiyab commented:
@thenamankumar Added a few comments. I'd like us to try and get them in as they vastly improve UX for little cost.

thenamankumar commented:
> @thenamankumar Added a few comments. I'd like us to try and get them in as they vastly improve UX for little cost.

- I updated the short cut to `Shift+Opt+P`.
- I also moved the description to below the title.
- Lets add the additional links like "New Prompt" later separately. I don't have more time this week. 
![CleanShot 2024-11-20 at 16 53 25@2x](https://github.com/user-attachments/assets/425f00b9-f372-4452-a61c-260f53e85ff6)

taiyab commented:
> > @thenamankumar Added a few comments. I'd like us to try and get them in as they vastly improve UX for little cost.
> 
> * I updated the short cut to `Shift+Opt+P`.
> * I also moved the description to below the title.
> * Lets add the additional links like "New Prompt" later separately. I don't have more time this week.
>   ![CleanShot 2024-11-20 at 16 53 25@2x](https://private-user-images.githubusercontent.com/22571395/388072032-425f00b9-f372-4452-a61c-260f53e85ff6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxMDY2NTksIm5iZiI6MTczMjEwNjM1OSwicGF0aCI6Ii8yMjU3MTM5NS8zODgwNzIwMzItNDI1ZjAwYjktZjM3Mi00NDUyLWE2MWMtMjYwZjUzZTg1ZmY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIwVDEyMzkxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4M2ZhODNkMzBlNzQ2YjBmMzg2ODFmODFjNzdjNDRiNWQwZGVjNjhkM2M5Y2VlM2QzNGI3YTZjZDFhNTZkNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3dx4u30GHnHqmDNZrz__hmoJtFUTQU-LK6M8HiQzR4Q)

Aweesome! No worries.

kalanchan commented:
Thanks all! I'm going to merge this as its a required feature we need to ship on Nov 25

thenamankumar commented on vscode/src/chat/chat-view/ChatsController.ts:
```diff
@@ -102,6 +103,26 @@ export class ChatsController implements vscode.Disposable {
+        const webviewPanelOrView =
+            this.panel.webviewPanelOrView || (await this.panel.createWebviewViewOrPanel())
+
+        setTimeout(
```
If there is no Cody panel opened, we need to create a new panel and render the webview. Thats way a delay in setting the editor state and more.
